src/stores/VideoChannelStore.ts

```diff
@@ -83,9 +79,19 @@ export default class VideoChannelStore extends AsyncStoreWithClient<null> {
 
     private activeChannel: ClientWidgetApi;
 
-    private _roomId: string;
-    public get roomId(): string { return this._roomId; }
-    private set roomId(value: string) { this._roomId = value; }
+    // This is persisted to localStorage so we can detect unclean disconnects
+    private _roomId = localStorage.getItem("mx_videoChannelRoomId");
+    public get roomId(): string | null { return this._roomId; }
+    private set roomId(value: string | null) {
+        this._roomId = value;
+        if (value === null) {
+            localStorage.removeItem("mx_videoChannelRoomId");
+        } else {
+            localStorage.setItem("mx_videoChannelRoomId", value);
+        }
+    }
```

Comment:
Temporary flags like this should go through the settings framework in the event we ever decide to support devices which don't have localstorage (a thing that's unlikely to happen, but likely enough to mention/defend for it)